### PR TITLE
 {ly5395} adding asset loading logic for procedural prefabs

### PR DESCRIPTION
**Pull request by jackalbe**
{ly5395} adding asset loading logic for procedural prefabs

* enabling the Prefab gem for tool work
* enabling prefab gem for AutomatedTesting
* AssetTypeInfoHandler for procedural prefab
* EnableCatalogForAsset for procedural prefab
* RegisterHandler for AssetManager

### CODE REVIEW

**Review by amzn-mike — APPROVED** [review]

File: Gems/Prefab/PrefabBuilder/PrefabGroup/ProceduralAssetHandler.cpp [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    }
+
+    const char* PrefabGroupAssetHandler::AssetTypeInfoHandler::GetBrowserIcon() const
+    {
+        return "Icons/Components/Box.png";
+    }
+
+    void PrefabGroupAssetHandler::AssetTypeInfoHandler::GetAssetTypeExtensions(AZStd::vector<AZStd::string>& extensions)
+    {
+        extensions.push_back("procprefab");
```
Comment (AMZN-stankowi) [inline Gems/Prefab/PrefabBuilder/PrefabGroup/ProceduralAssetHandler.cpp:68]: you should probably make this a const somewhere, so if we change it we only need to change it in one place.

File: Gems/Prefab/PrefabBuilder/PrefabGroup/ProceduralAssetHandler.cpp [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+        const AZ::Data::Asset<AZ::Data::AssetData>& asset,
+        AZStd::shared_ptr<AZ::Data::AssetDataStream> stream,
+        [[maybe_unused]] const AZ::Data::AssetFilterCB& assetLoadFilterCB)
+    {
+        using namespace AzToolsFramework::Prefab;
+
+        auto* proceduralPrefabAsset = asset.GetAs<ProceduralPrefabAsset>();
+        if (!proceduralPrefabAsset)
+        {
+            AZ_Error("prefab", false, "This should be a BlastChunksAsset type, as this is the only type we process!");
```
Comment (AMZN-stankowi) [inline Gems/Prefab/PrefabBuilder/PrefabGroup/ProceduralAssetHandler.cpp:133]: it looks like BlastChunksAsset is a copy and paste leftover

File: Gems/Prefab/PrefabBuilder/PrefabGroup/ProceduralAssetHandler.h [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/Asset/AssetManager.h>
+#include <PrefabGroup/PrefabGroup.h>
+#include <AzCore/Memory/Memory.h>
+#include <AzCore/Memory/SystemAllocator.h>
+#include <AzCore/std/containers/vector.h>
```
Comment (AMZN-stankowi) [inline Gems/Prefab/PrefabBuilder/PrefabGroup/ProceduralAssetHandler.h:14]: are you using all these includes here? I don't see a vector

Comment (jackalbe) [reply Gems/Prefab/PrefabBuilder/PrefabGroup/ProceduralAssetHandler.h:14]: void GetHandledAssetTypes(AZStd::vector<AZ::Data::AssetType>& assetTypes) override;

but I was able to optimize the headers

File: Gems/Prefab/PrefabBuilder/PrefabGroup/ProceduralAssetHandler.cpp [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+            return LoadResult::Error;
+        }
+
+        rapidjson::Document prefabDoc;
+        prefabDoc.CopyFrom(prefabData, prefabDoc.GetAllocator());
+        AZStd::string stringJson;
+        auto stringOutcome = AZ::JsonSerializationUtils::WriteJsonString(prefabDoc, stringJson);
+        if (stringOutcome.IsSuccess() == false)
+        {
+            AZ_Error("prefab", false, "Could not write to JSON string %s", stringOutcome.GetError().c_str());
```
Comment (AMZN-stankowi) [inline Gems/Prefab/PrefabBuilder/PrefabGroup/ProceduralAssetHandler.cpp:178]: this is the only result you're also emitting an AZ_Error with. Maybe each of the other error returns should also have an error message, to help track down specifically what went wrong?

**Review by AMZN-stankowi — APPROVED** [review]

**Pull request merged**